src/daft-connect/src/display.rs

```diff
@@ -0,0 +1,326 @@
+use std::fmt::Write;
+
+use daft_core::prelude::*;
+
+pub fn to_tree_string(schema: &Schema) -> eyre::Result<String> {
+    let mut output = String::new();
+    // Start with root
+    writeln!(&mut output, "root")?;
+    // Now print each top-level field
+    for (name, field) in &schema.fields {
+        print_field(&mut output, name, &field.dtype, /*nullable*/ true, 1)?;
+    }
+    Ok(output)
+}
+
+// A helper function to print a field at a given level of indentation.
+// level=1 means a single " |-- " prefix, level=2 means
+// " |    |-- " and so on, mimicking Spark's indentation style.
+// A helper function to print a field at a given level of indentation.
+fn print_field(
+    w: &mut String,
+    field_name: &str,
+    dtype: &DataType,
+    nullable: bool,
+    level: usize,
+) -> eyre::Result<()> {
+    let indent = if level == 1 {
+        " |-- ".to_string()
+    } else {
+        format!(" |{}-- ", "    |".repeat(level - 1))
+    };
+
+    let dtype_str = type_to_string(dtype);
+    writeln!(
+        w,
+        "{}{}: {} (nullable = {})",
+        indent, field_name, dtype_str, nullable
+    )?;
+
+    if let DataType::Struct(fields) = dtype {
+        for field in fields {
+            print_field(w, &field.name, &field.dtype, true, level + 1)?;
+        }
+    }
+
+    Ok(())
+}
+
+fn type_to_string(dtype: &DataType) -> String {
+    // We want a nice, human-readable type string.
+    // Spark generally prints something like "integer", "string", etc.
+    // We'll follow a similar style here:
+    match dtype {
+        DataType::Null => "null".to_string(),
+        DataType::Boolean => "boolean".to_string(),
+        DataType::Int8
+        | DataType::Int16
+        | DataType::Int32
+        | DataType::Int64
+        | DataType::UInt8
+        | DataType::UInt16
+        | DataType::UInt32
+        | DataType::UInt64 => "integer".to_string(), // Spark doesn't differentiate sizes
```

Comment:
there are many different int sizes

> 

    Numeric types
        ByteType: Represents 1-byte signed integer numbers. The range of numbers is from -128 to 127.
        ShortType: Represents 2-byte signed integer numbers. The range of numbers is from -32768 to 32767.
        IntegerType: Represents 4-byte signed integer numbers. The range of numbers is from -2147483648 to 2147483647.
        LongType: Represents 8-byte signed integer numbers. The range of numbers is from -9223372036854775808 to 9223372036854775807.
        FloatType: Represents 4-byte single-precision floating point numbers.
        DoubleType: Represents 8-byte double-precision floating point numbers.
        DecimalType: Represents arbitrary-precision signed decimal numbers. Backed internally by java.math.BigDecimal. A BigDecimal consists of an arbitrary precision integer unscaled value and a 32-bit integer scale.

Reply:
done